scipy/stats/_continuous_distns.py

```diff
@@ -6294,6 +6294,127 @@ def _stats(self, a):
+        # the methods to determine the shape, location, and scale.
+        def fit_with_multiple_funcs(fshape, floc, fscale, data, get_u, get_s,
+                                    fixed_shape_only, all_free):
+            if floc is not None:
+                # `floc` is fixed, `fshape` and `fscale` may be deterermined
+                # through analytical equations
+                if floc > np.min(data):
+                    # a constraint to fitting is that the location < np.min
+                    raise ValueError("Invalid values in `data`.  Maximum"
+                                     " likelihood estimation with 'powerlaw'"
+                                     " requires loc < min(data).")
+                if fscale is None:
+                    fscale = get_s(data, floc)
+                else:
+                    if fscale + floc <= np.max(data):
+                        # fitting with fixed scale and fixed location requires
+                        # that loc + scale > max(data)
+                        raise ValueError("Invalid values in `data`.  Maximum"
+                                         " likelihood estimation with"
+                                         " 'powerlaw' requires loc + scale >"
+                                         " max(data).")
+                if fshape is None:
+                    fshape = get_a(floc, fscale, data)
+                return fshape, floc, fscale
+            elif fscale is not None:
+                # When the scale is fixed, the location can be analytically
+                # determined.
+                floc = get_u(data, fscale)
+                if fshape is None:
+                    fshape = get_a(floc, fscale, data)
+                return fshape, floc, fscale
+            elif fshape is not None:
+                return fixed_shape_only(data, get_a, get_u, get_s, fshape)
+            else:
+                return all_free(data, get_a, get_u, get_s)
+
+        # fitting under the assumption that `fshape < 1` is fast and completely
+        # analytical, so it is performed first.
+        res = fit_with_multiple_funcs(fshape, floc, fscale, data, get_u_a_lt,
+                                      get_s_a_lt, fixed_shape_only,
+                                      all_free_lt)
+        # if the assumption that the shape < 1 is not consistent with the
+        # result of the fitting, try to fit under the assumption that the
+        # shape is >= 1.
+        if (not np.allclose(res[0], 1)) and (res[0] > 1 or res[0] is None):
+
+            res1 = fit_with_multiple_funcs(fshape, floc, fscale, data,
+                                          get_u_a_gt, get_s_a_gt,
+                                          all_free_gt, all_free_gt)
+            if res1[0] is not None:
+                # make sure that this was successful, otherwise, revert to
+                # other determination.
+                res = res1
+
+        return res
```

Comment:
This got a bit messy because a few cases came up where when `a` is approximately 1, the fit under the assumption that a < 1 doesn't give a result close to 1 or less than one, so then it checks the method where a > 1. However, that one sometimes fails (returns None), in which case it reverts back to the fit from the assumption that a < 1.